### ISSUE TITLE
Fix issue with returning (legacy) subdomain usernames with addresses endpoint

### ISSUE THREAD

**Issue by markmhx** [blockstack/stacks-blockchain-api#713]
I'd expect the following call to return "markhendrickson.id.blockstack" (which was registered before Stacks 2.0) under "names", but instead the response is empty:

https://stacks-node-api.mainnet.stacks.co/v1/addresses/stacks/SPG6XHZVNEEXTCDX634RGDHJ8X1R5C1VYH6WXR7P

![image](https://user-images.githubusercontent.com/28991/129862204-c4f6b72e-fa2c-42da-b3c8-9a7031590e68.png)

![image](https://user-images.githubusercontent.com/28991/129862292-b3ab7e85-7f9d-43ac-9612-082116afc763.png)

[First discussed on Discord here](https://discord.com/channels/621759717756370964/621759718192447502/877251109104795648)

cc @zone117x

**Comment by zone117x** [comment]
@asimm241 do you have bandwidth to take this on? Seems like a regression (probably would have happened during microblock implementation) or like the BNS import process isn't running correctly.

**Comment by asimm241** [comment]
@markmhx  I can see `markhendrickson.id.blockstack` registered for `SP3CG3Y2BF1RVS4M3MNZKRAHVRW2PY0ZF8DJ63G4S `  https://stacks-node-api.mainnet.stacks.co/v1/names/markhendrickson.id.blockstack returns 
```
{
address: "SP3CG3Y2BF1RVS4M3MNZKRAHVRW2PY0ZF8DJ63G4S",
blockchain: "stacks",
last_txid: "",
resolver: "https://registrar.blockstack.org",
status: "registered_subdomain",
zonefile: "$ORIGIN markhendrickson.id.blockstack $TTL 3600 _http._tcp IN URI 10 1 "https://gaia.blockstack.org/hub/1LnRzVuVb2SmMamQZQL7d4oFkbZCkPbcd9/profile.json" ",
zonefile_hash: "d4647919699674659283063bc4460e9ca474cfbe"
}
```

I can not find this address `SPG6XHZVNEEXTCDX634RGDHJ8X1R5C1VYH6WXR7P` anywhere in the legacy data.

This is the BTC address `1LnRzVuVb2SmMamQZQL7d4oFkbZCkPbcd9` associated with `markhendrickson.id.blockstack` which translates to `SP3CG3Y2BF1RVS4M3MNZKRAHVRW2PY0ZF8DJ63G4S`.

where does this come from `SPG6XHZVNEEXTCDX634RGDHJ8X1R5C1VYH6WXR7P`?

cc: @zone117x

**Comment by agraebe** [comment]
@aulneau @zone117x any ideas on the last comment?

**Comment by aulneau** [comment]
> @aulneau could the differences that @asimm241 sees above be related to our modified approach to derivation paths for addresses in the web wallet as implemented earlier this year for Stacks 2.0?

yes this is likely what the issue is

**Comment by zone117x** [comment]
Looks like this issue in client repo(s): https://github.com/blockstack/stacks.js/issues/1105
Closing this API issue
